### PR TITLE
Replace WinForms crash dialog with SDL_ShowMessageBox.

### PR DESCRIPTION
**Pull request by pchote**
This PR aims to solve another issue facing #17989, that .net core refuses to build anything that uses winforms on non-windows platforms.

This will also resolve https://github.com/OpenRA/OpenRAModSDK/issues/123 once the SDK is updated to use the newer engine.

Old dialog:
Screenshot 2020-09-02 at 18 07 10

New dialog:
Screenshot 2020-09-02 at 18 17 59

This includes a minimal wrapper that pinvokes directly into the sdl2.dll because I couldn't find any documentation on how to allow `csc` to compile code that depends on netstandard that doesn't involve hardcoded filepaths. We should be able to replace this with SDL2-CS in #17989.

Test builds: https://github.com/pchote/OpenRA/releases/tag/devtest-20200902

### CODE REVIEW

File: packaging/windows/WindowsLauncher.cs.in [hunk 1]
```diff
@@ -11,15 +11,126 @@
...
 using System.Linq;
 using System.Media;
 using System.Reflection;
-using System.Windows.Forms;
+using System.Runtime.InteropServices;
+using System.Text;
 
 namespace OpenRA
 {
+	// Minimal SDL wrapper to work around SDL2-CS requiring netstandard2.0
```
Comment (pchote) [inline packaging/windows/WindowsLauncher.cs.in:23]: If we want to go ahead with this approach I will need to reword this to clearly attribute SDL2-CS and its license.

**Comment by teinarss** [comment]
Another solution would to use https://avaloniaui.net/

**Comment by pchote** [comment]
As discussed on IRC, we'll do these changes in #17989 to avoid the uglyness of the SDL2-CS workaround.
